### PR TITLE
[Database metrics] Add function for easily testing many datasets

### PR DESCRIPTION
**Pull request by chinandrew**
Currently branched off `add_rows` since it needs some of those functions. Also the code is all prototype and I haven't actually run it yet.

#### Summary of Changes:
- Add a function that lets people easily submit a number of datasets and queries to run and capture metrics for.

The output format could get a bit messy due to the number of options and how far the dicts could get nested. The append_data option also adds a complication, since if it's True you'll want to capture either the order or the preceding datasets, and dicts can't have lists as keys and are unordered. I've implemented a fairly basic format right now with just a dict of lists, which hopefully is easy to extract and plot. I can imagine a number of other implementations that might all work fine though. @krivard any suggestions?

#### Examples
Measure metrics for load and metadata of usa-facts and fb-survey independently.
```
measure_database([("fb-survey", "*"), ("usa-facts", "*")],
                 client,
                 container)
```
Outputs something like 
```
# `<metrics>` is currently a dict
{
  "load": [<metrics>, <metrics>],
  "meta": [<metrics>, <metrics>]
}
```
Example call to measure metrics for load, metadata, and a query on March and March+April usa-facts data.
```
query1 = {"source": "covidcast",
          "data_source": "usa-facts",
          "signal": "deaths_incidence_num",
          "time_type": "day",
          "geo_type": "county",
          "time_values": "20200301-2020501",
          "geo_value": "*"}
measure_database([("usa-facts", "202003*_county*"), ("usa-facts", "202004*_county*")],
                 client,
                 container,
                 [query1],
                 append_data=True)
```
Output. Note the user will have to remember the append_data=True since the output doesn't note that. I guess I could just add a key for that..:
```
{
  "load": [<metrics>, <metrics>],
  "meta": [<metrics>, <metrics>],
  "query1": [<metrics>, <metrics>]
}
```

### CODE REVIEW

**Review by krivard — APPROVED** [review]
nice work. parallel lists is a reasonable approach for storing the output, and easy enough to pivot with `zip` if someone needs it organized the other way.

`append_data` was not immediately obvious, but the docstring makes it clear enough. I like the idea of including a key in the return dict containing the measurement configuration.

**Comment by chinandrew** [comment]
Made a few adjustments that I've yet to push, but here's a toy example using current progress to plot something:
```
measurements = measure_database([("usa-facts", "*202003*_county_deaths*"), 
                                 ("usa-facts", "*202004*_county_deaths*"),
                                 ("usa-facts", "*202005*_county_deaths*")],
                                client,
                                db_container,
                                append_datasets=True)

for action, metric_list in measurements.items():
    if action == "append_datasets":  # need to figure this case out
        continue
    rows = [i["table_rows"] for i in metric_list]
    disk = [i["db_disk_usage_mb"] for i in metric_list]
    runtime = [i["runtime"] for i in metric_list]
    memory = [i["memory_usage_mb"] for i in metric_list]

    fig, ax = plt.subplots(1,3)
    fig.suptitle(f"{action} metrics")
    ax[0].plot(rows, disk, linestyle='--', marker='o')
    ax[0].set_xlabel("rows")
    ax[0].set_ylabel("disk usage (MB)")

    ax[1].plot(rows, runtime, linestyle='--', marker='o')
    ax[1].set_xlabel("rows")
    ax[1].set_ylabel("runtime (s)")

    ax[2].plot(rows, memory, linestyle='--', marker='o')
    ax[2].set_xlabel("rows")
    ax[2].set_ylabel("peak memory (MB)")
```

Screenshot from 2020-11-17 15-25-56

File: tests/database_metrics/test_monitor.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            ],
+            "meta": [
+                {"table_rows": 5, "db_disk_usage_mb": 6., "runtime": 7, "memory_usage_mb": 8},
+                {"table_rows": 1, "db_disk_usage_mb": 3., "runtime": 5, "memory_usage_mb": 7}
+            ],
+            "query0": [
+                {"table_rows": 9, "db_disk_usage_mb": 10., "runtime": 11, "memory_usage_mb": 12},
+                {"table_rows": 3, "db_disk_usage_mb": 6., "runtime": 9, "memory_usage_mb": 12}
+            ],
+        }
```
Comment (chinandrew) [inline tests/database_metrics/test_monitor.py:47]: example output

**Comment by chinandrew** [comment]
@krivard ready for proper review after adding tests and a few of the things discussed.

**Review by krivard — APPROVED** [review]
lgtm, nice work

**Pull request merged**